test/distributed/fsdp/test_fsdp_state_dict.py

```diff
@@ -274,6 +276,42 @@ def test_state_dict_load_into_local_module(self):
         for fsdp_param, local_param in zip(fsdp_params, local_params):
             self.assertEqual(fsdp_param, local_param)
 
+    @skip_if_lt_x_gpu(2)
+    @parametrize("double_nest", [True])
+    def test_state_dict_skip_module(self, double_nest):
+        torch.cuda.set_device(self.rank)
+
+        def _create_module():
+            LINEAR_SKIP = "linear_skip"
+            with enable_wrap(wrapper_cls=FSDP):
+                module = SkipModel(double_nest=double_nest)
+                # Full name of linear_skip param tensors in SkipModel, as would be
+                # stored in checkpoint.
+                linear_skip_tensor_names = [
+                    k for k in dict(module.named_parameters()).keys()
+                    if LINEAR_SKIP in k
+                ]
+                # skip SkipModule
+                linear_skip = getattr(module, LINEAR_SKIP)
+                delattr(module, LINEAR_SKIP)
+                # Wrap FSDP
+                fsdp = wrap(module)
+                # reattach
+                setattr(module, LINEAR_SKIP, linear_skip)
+                return fsdp, linear_skip_tensor_names
+
+        fsdp, linear_skip_tensor_names = _create_module()
+        # Run a forward pass
+        inp = torch.randn((1, 10), device=torch.cuda.current_device())
+        loss = fsdp(inp)
+        loss.sum().backward()
+
+        state_dict = fsdp.state_dict()
```

Comment:
I think the overall goal to resolve the issue is to ensure that this can be loaded into a local module, I can try this though and add a follow up change. Or feel free to do so if you have the time.